### DISCUSSION TITLE
[Bug]: Failed to deploy Kafka : waiting for first consumer to be created before binding

### DISCUSSION THREAD

**Discussion by kishorer00** [strimzi/strimzi-kafka-operator#8380] [General]
### Bug Description

We are unable to deploy Kafka using strimzi-kafka-operator version ( 0.26.1 and 0.31.1 )on AWS EKS CLuster 1.25.
Its able deploy the pods for  strimzi-kafka-operator but not able to deploy the Kafka.

Note : Able to deploy Kafka using strimzi-kafka-operator version 0.34.0 

### Steps to reproduce

- Deploy strimzi-kafka-operator version ( 0.26.1 (OR) 0.31.1 )
- Deploy kafka with default JBOD and PVC settings.
- Deployment fail with Error " waiting for first consumer to be created before binding"
- No PODs will be deployed for Kafka and zookeeper
+++++++++++++++++++++++++++++++++++++++
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 200Gi
        deleteClaim: false
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      log.retention.hours: 168
  zookeeper:
    replicas: 1
    storage:
      type: persistent-claim
      size: 20Gi
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}
++++++++++++++++++++++++++++++++++

### Expected behavior

Kafka and zookeeper deployed without any issues

### Strimzi version

0.31.0

### Kubernetes version

1.25

### Installation method

Helm charts and kubectl files 

### Infrastructure

AWS EKS

### Configuration files and logs

Error : -

# kubectl describe pvc data-test-cluster-zookeeper-0 -n test
Name:          data-test-cluster-zookeeper-0
Namespace:     test
StorageClass:  gp2
Status:        Bound
Volume:        pvc-367864ef-1114-4521-a188-3a95121113a0
Labels:        app.kubernetes.io/instance=test-cluster
               app.kubernetes.io/managed-by=strimzi-cluster-operator
               app.kubernetes.io/name=zookeeper
               app.kubernetes.io/part-of=strimzi-test-cluster
               strimzi.io/cluster=test-cluster
               strimzi.io/kind=Kafka
               strimzi.io/name=test-cluster-zookeeper
Annotations:   pv.kubernetes.io/bind-completed: yes
               strimzi.io/delete-claim: false
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      20Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Used By:       test-cluster-zookeeper-0
Events:
  Type    Reason                 Age   From                         Message
  ----    ------                 ----  ----                         -------
  Normal  WaitForFirstConsumer   27m   persistentvolume-controller  waiting for first consumer to be created before binding

### Additional context

Able to deploy Kafka using strimzi-kafka-operator version 0.34.0

**Comment by scholzj** [comment]
You will have to format the YAMLs as code to make it clear what the structure is. Without that it is not readable. You would also need to share the logs.

**Reply by kishorer00** [reply]
Here ,i am attaching the YAML files for strimzi-kafka-operator and Kafka
[
[kafka-dry-run.txt](https://github.com/strimzi/strimzi-kafka-operator/files/11211018/kafka-dry-run.txt)
[strimzi-dry-run.txt](https://github.com/strimzi/strimzi-kafka-operator/files/11211019/strimzi-dry-run.txt)
](url)

**Reply by scholzj** [reply]
Great ... but the logs are needed as well.

**Comment by georgejdli** [comment]
I had the same issue and the operator pod logs were not helpful.
I ended up restarting the operator pod and then the zookeeper pods came online.

**Comment by kehao95** [comment]
`waiting for first consumer to be created before binding`
Usually the will happen when the k8s scheduler failed to scale your pod to any one of your nodes. You may find more info by describe the pending pods.
